### PR TITLE
chore: reduce dependencies and fix bugs in package build

### PR DESCRIPTION
**Pull request by lucagrementieri**
This PR shows the changes with respect to the upstream version needed to reduce dependencies and to compile the package correctly.

I tried to reduce the amount of changes to the minimum, to allow for the possibility of following updates from the upstream branch, so I used comments instead of deletion.

After the merge I can assign a tag to this version on master in order to ensure we keep track of it for the next upgrades.

Detailed explanation of changes:
- Many imports are removed because they are not used in the file.
- Some unused functions are commented because they are not used in our code.
- All public datasets are not imported because they are not used in our code.
- The files in `utils/optimization` are copied from the `tools/train_utils/optimization` without any modification because they are used in our production code. The `tools` folder is a sibling folder of the package folder so it's external to the package, so I don't understand how these functions could be accessed before.
- Some new dependencies have appeared because they were hidden sub-dependencies that are now needed explicitly.

### CODE REVIEW

File: pcdet/utils/common_utils.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 import random
 import shutil
 import subprocess
-import SharedArray
 
+# import SharedArray
```
Comment (bugra-yilmaz) [inline pcdet/utils/common_utils.py:8]: Are we getting rid of the infamous warning? 😄

**Review by bugra-yilmaz — APPROVED** [review]
I don't want to block the PR and also not very familiar with the downstream implications at the moment but why didn't you just remove the lines instead of commenting if we can go back to this version when we want?

File: pcdet/utils/common_utils.py (hunk 1 above)
Comment (lucagrementieri) [reply pcdet/utils/common_utils.py:8]: Yes, it's already like this in the latest OSS version!

**Comment by lucagrementieri** [comment]
@bugra-yilmaz I decided to do not delete functions because the deletion creates more Git conflicts than the addition of multiline comments. For example, if the commented function is updated upstream, if we merge the latest upstream version we obtain no conflict. If instead the function is removed, every modification causes conflicts.

Another reason is that comments are easily reversible if we need to use and activate more features from the same package.

**Review by mylanpromaton — APPROVED** [review]
Great! Looks good to me :)

**Pull request merged**